File: jupyter2/ps/build_system/readxml2/jl/readxml2/readxml.jl
import LightXML

function parse(filename)
    doc = LightXML.parse_file(filename)
    return doc
end

function height(node)
    if LightXML.is_elementnode(node)
        max_height = 0
        for child in LightXML.child_nodes(node)
            d = height(child)
            if d > max_height
                max_height = d
            end
        end
        max_height + 1
    else
        return 0
    end
end

File: jupyter2/ps/build_system/readxml2/go/readxml2/readxml.go
package main
import (
	"github.com/subchen/go-xmldom"
)

func parse(filename string) *xmldom.Document {
	node := xmldom.Must(xmldom.ParseFile(filename))
	return node
}

func height(node *xmldom.Node) int {
	if node == nil {
		return 0
	} else {
		max := 0
		for _, child := range node.Children {
			d := height(child)
			if d > max {
				max = d
			}
		}
		return max + 1
	}
}

